sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -41,6 +42,11 @@ case class Mode(
     this(child, 0, 0, Some(reverse))
   }
 
+  final lazy val collationId: Int = child.dataType match {
+    case c: StringType => c.collationId
+    case _ => CollationFactory.UTF8_BINARY_COLLATION_ID
+  }
```

Comment:
I see, child is just Expression in Mode

in this case, I would say remove final lazy val collationId (since it's only used in 1 place anyways), and just do pattern matching below

Reply:
@uros-db done